src/infrasys/system.py

```diff
@@ -744,6 +744,7 @@ def list_child_components(
         component: Component
         component_type: Optional[Type[Component]]
             Filter the returned list to components of this type.
+            If the type has subclasses, all subclasses will be included.
```

Comment:
This functionality is predicated on the paradigm where a 'concrete' type does not have subclasses. I'm not sure that we can enforce that, but we do discuss it in the documentation.